src/models/BanList.ts

```diff
@@ -281,6 +286,71 @@ class BanList extends EventEmitter {
         this.emit('BanList.update', this, changes);
         return changes;
     }
+
+    /**
+     * Inform the `BanList` about a new event from the room it is modelling.
+     * @param event An event from the room the `BanList` models to inform an instance about.
+     */
+    public updateForEvent(event: { event_id: string }): void {
```

Comment:
I have the intuition that, in some future iteration, we could rewrite some of BanList to process events as they are received, rather than calling the entire `updateList()` once in a while. I could be wrong.

Reply:
The reason I haven't just done it naively is because there's at least one difficult situation to solve where if the bot goes down for awhile and the sync token is relatively old, even though when Mjolnir starts it builds lists with the latest rules, as Mjonir catches up to the present with sync it will reapply older rules before it gets back to the present.  Which is problematic if say someone has issued and redacted the same set of rules in that time period

Reply:
Good point.